you-should-use.plugin.zsh

```diff
@@ -1,14 +1,56 @@
 #!/bin/zsh
 
 function ysu_message() {
+  local BOLD='\033[1m'
+  local NONE='\033[00m'
   echo "${BOLD}Found existing alias for \"$1\". You should use: \"$2\"${NONE}"
 }
 
 
-function _check_aliases() {
+function ysu_global_message() {
   local BOLD='\033[1m'
-  local RED='\e[31m'
   local NONE='\033[00m'
+  echo "${BOLD}Found existing global alias for \"$1\". You should use: \"$2\"${NONE}"
+}
+
+
+# Prevent command from running if hardcore mode enabled
+function _check_ysu_hardcore() {
+  if [[ "$YSU_HARDCORE" = 1 ]]; then
+      local RED='\e[31m'
+      local BOLD='\033[1m'
+      local NONE='\033[00m'
+      echo "${BOLD}${RED}You Should Use hardcore mode enabled. Use your aliases!${NONE}"
+      kill -s INT $$
+  fi
+}
+
+
+function _check_global_aliases() {
+  local found=false
+  alias -g | while read entry; do
+    local tokens=("${(@s/=/)entry}")
+    local k="${tokens[1]}"
+    local v="${tokens[2]}"
```

Comment:
Tried a few tests and this seems to cover all the cases :) http://zsh.sourceforge.net/Guide/zshguide05.html

```
$ alias -g test="hello ' world"
$ alias -g                     
NE='2>/dev/null'
test='hello '' world'
$ alias -g | while read entry; do                    
> local tokens=("${(@s/=/)entry}")
> echo ${(Q)tokens[2]} 
> done        
2>/dev/null
hello ' world
```